modules/common/src/main/scala/surge/internal/akka/cluster/Shard.scala

```diff
@@ -35,13 +35,15 @@ object Shard {
   sealed trait ShardMessage
 
   def props[IdType](shardId: String, regionLogicProvider: PerShardLogicProvider[IdType], extractEntityId: PartialFunction[Any, IdType])(
-      implicit tracer: Tracer): Props = {
-    Props(new Shard(shardId, regionLogicProvider, extractEntityId)(tracer))
+      implicit tracer: Tracer,
+      ec: ExecutionContext): Props = {
+    Props(new Shard(shardId, regionLogicProvider, extractEntityId)(tracer, ec))
   }
 }
 
 class Shard[IdType](shardId: String, regionLogicProvider: PerShardLogicProvider[IdType], extractEntityId: PartialFunction[Any, IdType])(
-    implicit val tracer: Tracer)
+    implicit val tracer: Tracer,
+    ec: ExecutionContext)
```

Comment:
Does this actor actually use an ExecutionContext anywhere? In general my preference for the EC to use within an actor is the actors dispatcher. As you performance tune or separate out actors to their own dispatcher it makes it much easier to reason about which actors are using what and ensuring things are isolated at that level.

Reply:
Makes sense, I'll look into removing the ec param.